gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1266 @@
+    completeZone.setDnsName(zone.getDnsName());
+    completeZone.setDescription(zone.getDescription());
+    completeZone.setNameServerSet(zone.getNameServerSet());
+    completeZone.setCreationTime(ISODateTimeFormat.dateTime().withZoneUTC()
+        .print(System.currentTimeMillis()));
+    completeZone.setId(BigInteger.valueOf(Math.abs(ID_GENERATOR.nextLong() % Long.MAX_VALUE)));
+    completeZone.setNameServers(randomNameservers());
+    ZoneContainer zoneContainer = new ZoneContainer(completeZone);
+    // create the default NS and SOA records
+    zoneContainer.dnsRecords().set(defaultRecords(completeZone));
+    // place the zone in the data collection
+    ProjectContainer projectContainer = findProject(projectId);
+    ZoneContainer oldValue = projectContainer.zones().putIfAbsent(
+        completeZone.getName(), zoneContainer);
+    if (oldValue != null) {
+      return Error.ALREADY_EXISTS.response(String.format(
+          "The resource 'entity.managedZone' named '%s' already exists", completeZone.getName()));
+    }
+    // now return the desired attributes
+    ManagedZone result = OptionParsers.extractFields(completeZone, fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          String.format("Error when serializing managed zone %s.", result.getName()));
+    }
+  }
+
+  /**
+   * Creates a new change, stores it, and if delayChange > 0, invokes processing in a new thread.
+   */
+  Response createChange(String projectId, String zoneName, Change change, String... fields) {
+    ZoneContainer zoneContainer = findZone(projectId, zoneName);
+    if (zoneContainer == null) {
+      return Error.NOT_FOUND.response(String.format(
+          "The 'parameters.managedZone' resource named %s does not exist.", zoneName));
+    }
+    // check that the change to be applied is valid
+    Response response = checkChange(change, zoneContainer);
+    if (response != null) {
+      return response;
+    }
+    // start applying
+    Change completeChange = new Change(); // copy to avoid side effects
+    if (change.getAdditions() != null) {
+      completeChange.setAdditions(ImmutableList.copyOf(change.getAdditions()));
+    }
+    if (change.getDeletions() != null) {
+      completeChange.setDeletions(ImmutableList.copyOf(change.getDeletions()));
+    }
+    /* We need to set ID for the change. We are working in concurrent environment. We know that the
+    element fell on an index between 0 and maxId, so we will reset all IDs in this range (all of
+    them are valid for the respective objects). */
+    ConcurrentLinkedQueue<Change> changeSequence = zoneContainer.changes();
+    changeSequence.add(completeChange);
+    int maxId = changeSequence.size();
+    int index = 0;
+    for (Change c : changeSequence) {
+      if (index == maxId) {
+        break;
+      }
+      c.setId(String.valueOf(++index)); // indexing from 1
```

Comment:
I've convinced myself that there's no concurrency situation that will lead to elements that have been given IDs to get different IDs.  If we ignore elements that haven't yet been given an ID when listing changes (see comment about unassigned IDs/NumberFormatException), I think that page tokens will always be valid.  Do you agree with that analysis?